packages/intl-datetimeformat/src/core.ts

```diff
@@ -414,8 +414,11 @@ function initializeDateTimeFormat(
     if (matcher === 'basic') {
       bestFormat = basicFormatMatcher(opt, formats);
     } else {
-      opt.hour12 =
-        internalSlots.hourCycle === 'h11' || internalSlots.hourCycle === 'h12';
+      if (opt.hour !== undefined) {
```

Comment:
I think you'd have to expand this to all time-based attributes like `minutes`/`seconds`..., just like `toDateTimeOptions`

Reply:
I'm referring to https://github.com/formatjs/formatjs/blob/main/packages/intl-datetimeformat/src/core.ts#L508 so `['hour', 'minute', 'second']` it looks like